plugins/builds/index.js

```diff
@@ -244,9 +242,9 @@ async function triggerNextJobs(config, app) {
 
         // Skip trigger process if createExternalEvent fails
         if (externalEvent) {
-            for (const [nextJobName, nextJob] of Object.entries(joinedPipeline.jobs)) {
-                const nextJobId = nextJob.id || (await getJobId(nextJobName, currentPipeline.id, jobFactory));
-                const { isVirtual: isNextJobVirtual, stageName: nextJobStageName } = nextJob;
+            for (const [nextJobName, nextJobInfo] of Object.entries(joinedPipeline.jobs)) {
+                const nextJob = await getJob(nextJobName, joinedPipelineId, jobFactory);
```

Comment:
I believe there was a long living bug which used `currentPipeline.id` instead of `joinedPipelineId` to get the job associated with external pipeline.
Probably the execution control never reached the bug as the `nextJob.id` would be always defined.